### PR TITLE
Component updates: Based on feedback from use cases

### PR DESCRIPTION
**Pull request by chromaticWaster**
### Description:

(*Link to bug/feature from [Issues](../issues) and how you resolve it*)

closes https://github.com/capitec/omni-components/issues/99
closes https://github.com/capitec/omni-components/issues/80

- Introduce clearable attribute to OmniFormElement to introduce functionality to clear components value that inherit from the OmniFormElement base class.
- Introduce dialog to Date picker and Select components for mobile modes this introduces a default overlay.
- Resolve issue with Keyboard component not updating value for Currency field correctly.
- Updated stories of components to have a Clear story.
- Resolved the positioning of the Color field component label.
- Removed clear capability on the Search field component.
- Resolved label position of Currency field prefix story.
- Added max length implementation to the currency field to limit the value of the input to a trillion cent value

### Screenshots (if appropriate):

**Test results**

image

### Clear functionality

- Changes to the attribute name or story adjustments are welcomed

Note this is illustrated for components that work with different datatypes.

**Text Field**

Text field clear

**Date Picker**

Date picker clear

**Select**

Select clear

**Number Field**

Number field clear

### Color Field

**Before** 
image

**After**
image

### Dialogs for Date picker and Select

**Date Picker Component**

**Before** 

Date picker old implementation

**After**

Date picker dialog

**Select Component**

**Before**

Select old implementation

**After**

Select dialog

### Max length for Currency field

Before 

before maxlength

After

after maxlength

### All Submissions:

* [x] I have followed the [contribution guidelines](https://github.com/capitec/omni-components/blob/develop/CONTRIBUTING.md) for this project.
* [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.

### Changes to Core Features:

* [x] I have added an explanation of what my changes do and why I would like to include them.
* [x] I have written new tests for my core changes, as applicable.
* [x] I have successfully ran tests with my changes locally.
* [x] I have added/updated docs, as applicable.

### CODE REVIEW

File: src/core/OmniFormElement.ts [hunk 1]
```diff
@@ -115,6 +122,31 @@ export class OmniFormElement extends OmniElement {
      */
     @property({ type: Boolean, reflect: true }) disabled = false;
 
+    /**
+     * Toggles the ability to clear the value of the component.
+     * @attr
+     */
+    @property({ type: Boolean, reflect: true }) clearable = false;
+
+    protected _onClearClick(e: MouseEvent) {
```
Comment (BOTLANNER) [inline src/core/OmniFormElement.ts:131]: Should this not be named to indicate the action being performed. e.g. `clearValue`

File: src/core/OmniInputStories.ts [hunk 2]
```diff
@@ -156,6 +157,42 @@ export const SuffixStory = <T extends HTMLElement, U extends BaseArgs>(tagName:
...
 };
 
+export const ClearableStory = <T extends HTMLElement, U extends BaseArgs>(
+    tagName: string,
+    inputValue: string | number | string[] = 'The input value'
+) => {
+    const Clearable: ComponentStoryFormat<U> = {
+        render: (args: U) =>
+            html`${unsafeHTML(
+                `<${tagName} data-testid="test-field" label="${ifNotEmpty(args.label)}" value="${args.value}" clearable=${
```
Comment (BOTLANNER) [inline src/core/OmniInputStories.ts:167]: Consider using a Boolean attribute for clearable.

File: src/search-field/SearchField.stories.ts [hunk 3]
```diff
@@ -53,21 +65,10 @@ export const Interactive: ComponentStoryFormat<BaseArgs> = {
         await userEvent.type(inputField, value);
 
         // TODO: Fix race conditions in tests
```
Comment (BOTLANNER) [inline src/search-field/SearchField.stories.ts:67]: Remove comment for TODO

File: src/select/Select.ts [hunk 4]
```diff
@@ -159,14 +160,32 @@ export class Select extends OmniFormElement {
         window.addEventListener('scroll', this._checkScreenDimensions.bind(this));
     }
 
-    _inputClick() {
+    _inputClick(e: Event) {
+        /*
```
Comment (BOTLANNER) [inline src/select/Select.ts:164]: Remove commented code.

File: src/select/Select.ts [hunk 5]
```diff
@@ -178,8 +197,21 @@ export class Select extends OmniFormElement {
     _togglePopup() {
         if (this._popUp) {
             this._popUp = false;
+            if (this._isMobile) {
+                //this._pickerContainer?.close();
```
Comment (BOTLANNER) [inline src/select/Select.ts:201]: Remove commented code

**Review by BOTLANNER — CHANGES_REQUESTED** [review]

File: src/core/OmniFormElement.ts (hunk 1 above)
Comment (chromaticWaster) [reply src/core/OmniFormElement.ts:131]: Updated to name to clearValue

File: src/select/Select.ts (hunk 5 above)
Comment (chromaticWaster) [reply src/select/Select.ts:201]: Updated

File: src/select/Select.ts (hunk 4 above)
Comment (chromaticWaster) [reply src/select/Select.ts:164]: Updated

File: src/search-field/SearchField.stories.ts (hunk 3 above)
Comment (chromaticWaster) [reply src/search-field/SearchField.stories.ts:67]: Updated

File: src/core/OmniInputStories.ts (hunk 2 above)
Comment (chromaticWaster) [reply src/core/OmniInputStories.ts:167]: Updated as per suggestion

**Review by BOTLANNER — APPROVED** [review]

**Pull request merged**